docs/configuration/listeners/lds.rst

```diff
@@ -0,0 +1,77 @@
+.. _config_listeners_lds:
+
+Listener discovery service
+==========================
+
+The listener discovery service (LDS) is an optional API that Envoy will call to dynamically fetch
+listeners. Envoy will reconcile the API response and add, modify, or remove known listeners
+depending on what is required.
+
+The semantics of listener updates are as follows:
+
+* Every listener must have a unique :ref:`name <config_listeners_name>`. If a name is not
+  provided, Envoy will create a UUID. Listeners that are to be dynamically updated should have a
+  unique name supplied by the management server.
+* When a listener is added, it will be "warmed" before taking traffic. For example, if the listener
+  references an :ref:`RDS <config_http_conn_man_rds>` configuration, that configuration will be
+  resolved and fetched before the listener is moved to "active."
```

Comment:
Sorry, didn't understand what you meant. No, we don't do that. The only warming we do is:
1) During startup, the cluster manager goes through a multi-phase init where it first initializes stats/dns clusters, then predefined SDS clusters. Then it initializes CDS, waits for one response, and does the same primary/secondary init of DNS clusters.
1a) If clusters use active health checking, we also do a single active HC round.
2) Once cluster manager init is done, LDS/RDS initialize. The server doesn't start running until there has been at least one response (or failure) for LDS/RDS requests.
3) If LDS itself returns something that needs an RDS response, we further wait until we have received one RDS response or failure.

The above is done so that in the hot restart case we never start taking traffic until we have had a chance to do a complete init pass which will mean that there should be zero disruption in traffic.

I'm actually going to add the above explanation to the docs somewhere. It will be in the PR update.